zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/LearnerMetricsTest.java

```diff
@@ -32,16 +34,30 @@
 import org.hamcrest.Matcher;
 import org.junit.After;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+@RunWith(Parameterized.class)
 public class LearnerMetricsTest extends QuorumPeerTestBase {
 
     private static final int TIMEOUT_SECONDS = 30;
     private static final int SERVER_COUNT = 4; // 1 observer, 3 participants
     private final QuorumPeerTestBase.MainThread[] mt = new QuorumPeerTestBase.MainThread[SERVER_COUNT];
     private ZooKeeper zk_client;
+    private boolean asyncSending;
+
+    public LearnerMetricsTest(boolean asyncSending) {
+        this.asyncSending = asyncSending;
+    }
+
+    @Parameterized.Parameters
+    public static Collection sendingModes() {
+        return Arrays.asList(new Object[][]{{true}, {false}});
+    }
 
     @Test
     public void testLearnerMetricsTest() throws Exception {
+        Learner.setAsyncSending(asyncSending);
```

Comment:
please reset the value in a @Before method
it is a static flag, so it will affect other tests

Reply:
I assume you mean "reset in @after?"